tests/unit/s2n_map_test.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+
+#include <s2n.h>
+#include <string.h>
+
+#include "utils/s2n_map.h"
+
+int main(int argc, char **argv)
+{
+    char keystr[sizeof("ffff")];
+    char valstr[sizeof("8192")];
+    struct s2n_map *map;
+    struct s2n_blob key;
+    struct s2n_blob val;
+
+    BEGIN_TEST();
+
+    EXPECT_NOT_NULL(map = s2n_map_new());
```

Comment:
The test cases look pretty good here, but you could also test the edge case of doing a lookup on an empty map.